### PR TITLE
Add new constructors to create SxgWorker from parsed data

### PR DESCRIPTION
**Pull request by quangIO**
This also follows some suggestions in #191.

### CODE REVIEW

File: sxg_rs/src/config.rs [hunk 1]
```diff
@@ -58,29 +59,52 @@ fn lowercase_all(names: BTreeSet<String>) -> BTreeSet<String> {
...
+        strip_request_headers: lowercase_all(input.strip_request_headers),
+        strip_response_headers: lowercase_all(input.strip_response_headers),
+        validity_url_dirname: to_url_prefix(&input.validity_url_dirname),
+        ..input
+    };
+    input
+}
+
 impl Config {
+    /// Create config from text
```
Comment (antiphoton) [inline sxg_rs/src/config.rs:76]:
```suggestion
    /// Creates config from text
```

File: sxg_rs/src/config.rs [hunk 2]
```diff
@@ -58,29 +59,52 @@ fn lowercase_all(names: BTreeSet<String>) -> BTreeSet<String> {
...
         let issuer_der = get_der_from_pem(issuer_pem, "CERTIFICATE")?;
-        let cert_url_dirname = to_url_prefix(&input.cert_url_dirname);
-        let reserved_path = to_url_prefix(&input.reserved_path);
-        let validity_url_dirname = to_url_prefix(&input.validity_url_dirname);
+
+        let config = Self::new_with_parsed_data(input, cert_der, issuer_der);
+
+        Ok(config)
+    }
+    /// Create new from parsed input and der
```
Comment (antiphoton) [inline sxg_rs/src/config.rs:86]:
```suggestion
    /// Creates new from parsed input and der
```

File: sxg_rs/src/config.rs [hunk 3]
```diff
@@ -58,29 +59,52 @@ fn lowercase_all(names: BTreeSet<String>) -> BTreeSet<String> {
...
+    pub fn new_with_parsed_data(
+        input: ConfigInput,
+        cert_der: Vec<u8>,
+        issuer_der: Vec<u8>,
+    ) -> Self {
+        let input = normalize_config_input(input);
+        Self::new_with_parsed_data_unchecked(input, cert_der, issuer_der)
+    }
+    /// Create a new config without normalizing input
+    pub fn new_with_parsed_data_unchecked(
```
Comment (antiphoton) [inline sxg_rs/src/config.rs:96]:
```suggestion
    /// Creates a new config without normalizing input
    pub fn new_with_parsed_and_normalized_data(
```

File: sxg_rs/src/lib.rs [hunk 4]
```diff
@@ -318,6 +322,12 @@ impl SxgWorker {
     }
 }
 
+impl From<Config> for SxgWorker {
+    fn from(config: Config) -> Self {
+        Self { config }
+    }
+}
+
```
Comment (antiphoton) [inline sxg_rs/src/lib.rs:330]: It is possible that `SxgWorker` will have more member variables other than `config` itself in the future, so I think we should not add this.

Comment (quangIO) [reply sxg_rs/src/lib.rs:330]: Perhaps I can change it to a new constructor like `new_with_config(config: Config)`?

Comment (antiphoton) [reply sxg_rs/src/lib.rs:330]: Yes.

**Review by antiphoton — DISMISSED** [review]

**Comment by quangIO** [comment]
Sorry, rusty fingers. I force-pushed new code to fix typo.

**Review by antiphoton — APPROVED** [review]

**Pull request merged**